src/main/java/org/jenkinsci/plugins/azure/AzurePublisherSettings.java

```diff
@@ -28,115 +28,21 @@
 import com.cloudbees.plugins.credentials.CredentialsScope;
 import com.cloudbees.plugins.credentials.impl.BaseStandardCredentials;
 import hudson.Extension;
-import hudson.FilePath;
-import hudson.remoting.VirtualChannel;
-import jenkins.security.MasterToSlaveCallable;
-import org.apache.commons.io.FileUtils;
+
 import org.kohsuke.stapler.DataBoundConstructor;
 
-import javax.annotation.Nonnull;
-import java.io.ByteArrayInputStream;
-import java.io.File;
 import java.io.IOException;
-import java.io.InputStream;
-import java.nio.charset.StandardCharsets;
 
 /**
  * @author <a href="mailto:nicolas.deloof@gmail.com">Nicolas De Loof</a>
  */
 public class AzurePublisherSettings extends BaseStandardCredentials {
 
-    public static final String SUBSCRIPTION_XPATH = "/PublishData/PublishProfile/Subscription";
-
-    private final String subscriptionId;
-
-    private final String subscriptionName;
-
-    private final String serviceManagementCert;
-
-    private final String serviceManagementUrl;
-
-/*
-    public AzurePublisherSettings(CredentialsScope scope, String id, String description, FileItem file, String fileName, String data) throws IOException {
-        super(scope, id, description);
-
-        XPath xp = XPathFactory.newInstance().newXPath();
-        InputSource ip = new InputSource(new ByteArrayInputStream(file.get()));
-        try {
-            Element node = (Element) xp.evaluate(SUBSCRIPTION_XPATH, ip, XPathConstants.NODE);
-            this.subscriptionId = node.getAttribute("Id");
-            this.subscriptionName = node.getAttribute("Name");
-            this.serviceManagementCert = node.getAttribute("ManagementCertificate");
-            this.serviceManagementUrl = node.getAttribute("ServiceManagementUrl");
-        } catch (XPathExpressionException e) {
-            throw new IOException("Invalid PublishSettings file", e);
-        }
-    }*/
-
     @DataBoundConstructor
-    public AzurePublisherSettings(CredentialsScope scope, String id, String description, String subscriptionId, String subscriptionName, String serviceManagementCert, String serviceManagementUrl) {
+    public AzurePublisherSettings(CredentialsScope scope, String id, String description, String subscriptionId,
+                                  String subscriptionName, String serviceManagementCert, String serviceManagementUrl) throws IllegalStateException {
```

Comment:
IllegalStateException => RuntimeException, does not require the signature change here.

Reply:
Agreed - its just left as a form of documentation.